SourceryFramework/Sources/Parsing/FileParser.swift

```diff
@@ -470,8 +480,12 @@ extension FileParser {
     }
 
     internal func parseVariable(_ source: [String: SourceKitRepresentable], definedIn: Type?, isStatic: Bool = false) -> Variable? {
-        guard let (name, _, accessibility) = parseTypeRequirements(source) else { return nil }
+        guard let (nameOrNil, _, accessibility) = parseTypeRequirements(source) else { return nil }
 
+        guard let name = nameOrNil else {
+            return nil
+        }
```

Comment:
why to have two separate guards if both result in the same result? shouldn't the first `guard let (name, _ accessibility) = ...` be enough? same in other methods.

Reply:
It's a personal preference that arised after discovery that joined guards severely impact performance of compilation. The difference is noticeable. I remember doing researches several times, each of them resulted in significantly improved performance of compilation, like 20-30%.

It's like preferences of avoiding structs, making everything final, avoiding concatenation of strings (a + b + c works much slower than "\(a)\(b)\(c)"), avoiding auto-typing here and there, splitting long functions into parts (which is a good thing anyway), etc. I'm not sure about marking all protocols as `class` for example (`protocol Something: class`), it's a standard we use, but I didn't research performance impact on it (it's possible that it affects performance, but mainly it's use to allow usage of weak referencing).

That thing about `guard` is very real, it can easily make compilation time of a function more than a second, which doesn't sound all that bad, but for big enough project it can make compilation time incredibly long.

I would join guard in this pull request though, because it will be more fitting to the codestyle of the project.